### PR TITLE
[chores, refactor] Enable torchscript tests internally, refactor

### PR DESCRIPTION
**Pull request by vedanuj**
Summary:
- Enable torchscript tests internally by enabling proxy
- Remove code duplication in torchscript tests

Differential Revision: D24213541

### CODE REVIEW

**Review by apsdehal — COMMENTED** [review]
You will need to adapt vilbert test now as well.

File: mmf/models/vilbert.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def __init__(self, config):
         self.key = nn.Linear(config.v_hidden_size, self.all_head_size)
         self.value = nn.Linear(config.v_hidden_size, self.all_head_size)
 
-        if self.dynamic_attention or torch.jit.is_scripting():
```
Comment (apsdehal) [inline mmf/models/vilbert.py:161]: This was not needed?

File: tests/models/test_mmbt.py [hunk 2]
```diff
@@ -33,34 +31,14 @@ def setUp(self):
...
-            model_output = self.finetune_model.model(test_sample_list)
-
-        script_model = torch.jit.script(self.finetune_model.model)
-        with torch.no_grad():
-            script_output = script_model(test_sample_list)
-
-        self.assertTrue(torch.equal(model_output["scores"], script_output["scores"]))
+        model = self.finetune_model.eval()
+        self.assertTrue(
+            test_utils.compare_torchscript_transformer_models(model, vocab_size=30255)
```
Comment (apsdehal) [inline tests/models/test_mmbt.py:40]: I would convert vocab_size to a constant such as BERT_VOCAB_SIZE etc.

**Review by apsdehal — APPROVED** [review]

File: mmf/models/vilbert.py (hunk 1 above)
Comment (vedanuj) [reply mmf/models/vilbert.py:161]: No it's not required.

**Pull request closed without merging**